### ISSUE TITLE
Better error handling when user does not provide argument for -e

### ISSUE THREAD

**Issue by yujunz** [pypa/pipenv#2383]
I am trying to install a local package for development purpose and encountered `AssertError`

<details><summary>$ python -m pipenv.help output</summary>

Pipenv version: `'2018.05.18'`

Pipenv location: `'/Users/yujunz/.pyenv/versions/2.7.13/lib/python2.7/site-packages/pipenv'`

Python location: `'/Users/yujunz/.pyenv/versions/2.7.13/bin/python'`

Other Python installations in `PATH`:

  - `2.7`: `/Users/yujunz/.pyenv/versions/2.7.13/bin/python2.7`
  - `2.7`: `/Users/yujunz/.pyenv/versions/2.7.13/bin/python2.7`
  - `2.7`: `/Users/yujunz/.pyenv/shims/python2.7`
  - `2.7`: `/usr/local/bin/python2.7`
  - `2.7`: `/usr/bin/python2.7`
  - `2.7`: `/usr/local/bin/python2.7`
  - `2.7`: `/usr/bin/python2.7`
  - `3.5`: `/Users/yujunz/.pyenv/shims/python3.5m`
  - `3.5`: `/Users/yujunz/.pyenv/shims/python3.5`
  - `3.6`: `/Users/yujunz/.pyenv/shims/python3.6`

  - `2.7.13`: `/Users/yujunz/.pyenv/versions/2.7.13/bin/python`
  - `2.7.13`: `/Users/yujunz/.pyenv/shims/python`
  - `2.7.10`: `/usr/bin/python`
  - `2.7.10`: `/usr/bin/python`
  - `2.7.13`: `/Users/yujunz/.pyenv/versions/2.7.13/bin/python2`
  - `2.7.13`: `/Users/yujunz/.pyenv/shims/python2`
  - `2.7.14`: `/usr/local/bin/python2`
  - `2.7.14`: `/usr/local/bin/python2`
  - `3.5.3`: `/Users/yujunz/.pyenv/shims/python3`

PEP 508 Information:

```
{'implementation_name': 'cpython',
 'implementation_version': '0',
 'os_name': 'posix',
 'platform_machine': 'x86_64',
 'platform_python_implementation': 'CPython',
 'platform_release': '17.6.0',
 'platform_system': 'Darwin',
 'platform_version': 'Darwin Kernel Version 17.6.0: Tue May  8 15:22:16 PDT 2018; root:xnu-4570.61.1~1/RELEASE_X86_64',
 'python_full_version': '2.7.13',
 'python_version': '2.7',
 'sys_platform': 'darwin'}
```

System environment variables:

  - `CPPFLAGS`
  - `GOPATH`
  - `LESS`
  - `http_proxy`
  - `COLORFGBG`
  - `LC_CTYPE`
  - `TERM_PROGRAM_VERSION`
  - `PYENV_DIR`
  - `PKG_CONFIG_PATH`
  - `LOGNAME`
  - `USER`
  - `HOME`
  - `PATH`
  - `VIRTUALENVWRAPPER_SCRIPT`
  - `ZSH`
  - `DISPLAY`
  - `TERM_PROGRAM`
  - `LANG`
  - `TERM`
  - `SHELL`
  - `VIRTUALENVWRAPPER_PYTHON`
  - `VIRTUALENVWRAPPER_LAZY_SCRIPT`
  - `PYENV_SHELL`
  - `SHLVL`
  - `SECURITYSESSIONID`
  - `ITERM_SESSION_ID`
  - `XPC_FLAGS`
  - `https_proxy`
  - `PYENV_VERSION`
  - `EDITOR`
  - `LDFLAGS`
  - `PYENV_VIRTUALENVWRAPPER_PYENV_VERSION`
  - `VIRTUALENVWRAPPER_VIRTUALENV_CLONE`
  - `COLORTERM`
  - `ARCHFLAGS`
  - `WORKON_HOME`
  - `TERM_SESSION_ID`
  - `XPC_SERVICE_NAME`
  - `MANPATH`
  - `SSH_AUTH_SOCK`
  - `Apple_PubSub_Socket_Render`
  - `ATLAS_TOKEN`
  - `PYTHONDONTWRITEBYTECODE`
  - `ITERM_PROFILE`
  - `LC_ALL`
  - `PIP_PYTHON_PATH`
  - `GPG_TTY`
  - `TMPDIR`
  - `VIRTUALENVWRAPPER_HOOK_DIR`
  - `VIRTUALENVWRAPPER_PROJECT_FILENAME`
  - `LSCOLORS`
  - `AUTOJUMP_SOURCED`
  - `PYENV_HOOK_PATH`
  - `PYENV_ROOT`
  - `__CF_USER_TEXT_ENCODING`
  - `no_proxy`
  - `PWD`
  - `AUTOJUMP_ERROR_PATH`
  - `VIRTUALENVWRAPPER_VIRTUALENV`
  - `BROWSER`
  - `PAGER`
  - `VIRTUALENVWRAPPER_WORKON_CD`
  - `COMMAND_MODE`

Pipenv–specific environment variables:

Debug–specific environment variables:

  - `PATH`: `/Users/yujunz/.pyenv/versions/2.7.13/bin:/usr/local/Cellar/pyenv/1.2.1/libexec:/Users/yujunz/Downloads/google-cloud-sdk/bin:/usr/local/opt/libressl/bin:/usr/local/opt/openssl/bin:/usr/local/opt/curl/bin:/Users/yujunz/.pyenv/shims:/usr/local/bin:/usr/bin:/bin:/usr/local/sbin:/usr/sbin:/sbin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin:/usr/local/go/bin:/Library/TeX/texbin:/Applications/Wireshark.app/Contents/MacOS:/Users/yujunz/Workspace/istio-0.6.0/bin:/Users/yujunz/go/bin:/Users/yujunz/.cargo/bin`
  - `SHELL`: `/bin/zsh`
  - `EDITOR`: `vim`
  - `LANG`: `en_US.UTF-8`
  - `PWD`: `/Users/yujunz/Workspace/pdm-cli`

---------------------------

Contents of `Pipfile` ('/Users/yujunz/Workspace/pdm-cli/Pipfile'):

```toml
[[source]]
url = "https://pypi.python.org/simple"
verify_ssl = true
name = "pypi"

[dev-packages]
pylint = "*"
pytest = "*"

[packages]
jmespath = "*"
"jinja2" = "*"
"1a2bcc9" = {path = "./../python-anyconfig"}

[requires]
python_version = "2.7"

```

Contents of `Pipfile.lock` ('/Users/yujunz/Workspace/pdm-cli/Pipfile.lock'):

```json
{
    "_meta": {
        "hash": {
            "sha256": "7a3500525ee36dd92ebe13ec2ca04bd3a477fe7506ab9a312ee50fbc883c42c1"
        },
        "pipfile-spec": 6,
        "requires": {
            "python_version": "2.7"
        },
        "sources": [
            {
                "name": "pypi",
                "url": "https://pypi.python.org/simple",
                "verify_ssl": true
            }
        ]
    },
    "default": {
        "1a2bcc9": {
            "path": "./../python-anyconfig"
        },
        "jinja2": {
            "hashes": [
                "sha256:74c935a1b8bb9a3947c50a54766a969d4846290e1e788ea44c1392163723c3bd",
                "sha256:f84be1bb0040caca4cea721fcbbbbd61f9be9464ca236387158b0feea01914a4"
            ],
            "index": "pypi",
            "version": "==2.10"
        },
        "jmespath": {
            "hashes": [
                "sha256:6a81d4c9aa62caf061cb517b4d9ad1dd300374cd4706997aff9cd6aedd61fc64",
                "sha256:f11b4461f425740a1d908e9a3f7365c3d2e569f6ca68a2ff8bc5bcd9676edd63"
            ],
            "index": "pypi",
            "version": "==0.9.3"
        },
        "markupsafe": {
            "hashes": [
                "sha256:a6be69091dac236ea9c6bc7d012beab42010fa914c459791d627dad4910eb665"
            ],
            "version": "==1.0"
        }
    },
    "develop": {
        "astroid": {
            "hashes": [
                "sha256:0ef2bf9f07c3150929b25e8e61b5198c27b0dca195e156f0e4d5bdd89185ca1a",
                "sha256:fc9b582dba0366e63540982c3944a9230cbc6f303641c51483fa547dcc22393a"
            ],
            "version": "==1.6.5"
        },
        "atomicwrites": {
            "hashes": [
                "sha256:240831ea22da9ab882b551b31d4225591e5e447a68c5e188db5b89ca1d487585",
                "sha256:a24da68318b08ac9c9c45029f4a10371ab5b20e4226738e150e6e7c571630ae6"
            ],
            "version": "==1.1.5"
        },
        "attrs": {
            "hashes": [
                "sha256:4b90b09eeeb9b88c35bc642cbac057e45a5fd85367b985bd2809c62b7b939265",
                "sha256:e0d0eb91441a3b53dab4d9b743eafc1ac44476296a2053b6ca3af0b139faf87b"
            ],
            "version": "==18.1.0"
        },
        "backports.functools-lru-cache": {
            "hashes": [
                "sha256:9d98697f088eb1b0fa451391f91afb5e3ebde16bbdb272819fd091151fda4f1a",
                "sha256:f0b0e4eba956de51238e17573b7087e852dfe9854afd2e9c873f73fc0ca0a6dd"
            ],
            "markers": "python_version == '2.7'",
            "version": "==1.5"
        },
        "configparser": {
            "hashes": [
                "sha256:5308b47021bc2340965c371f0f058cc6971a04502638d4244225c49d80db273a"
            ],
            "markers": "python_version == '2.7'",
            "version": "==3.5.0"
        },
        "enum34": {
            "hashes": [
                "sha256:2d81cbbe0e73112bdfe6ef8576f2238f2ba27dd0d55752a776c41d38b7da2850",
                "sha256:644837f692e5f550741432dd3f223bbb9852018674981b1664e5dc339387588a",
                "sha256:6bd0f6ad48ec2aa117d3d141940d484deccda84d4fcd884f5c3d93c23ecd8c79",
                "sha256:8ad8c4783bf61ded74527bffb48ed9b54166685e4230386a9ed9b1279e2df5b1"
            ],
            "markers": "python_version < '3.4'",
            "version": "==1.1.6"
        },
        "funcsigs": {
            "hashes": [
                "sha256:330cc27ccbf7f1e992e69fef78261dc7c6569012cf397db8d3de0234e6c937ca",
                "sha256:a7bb0f2cf3a3fd1ab2732cb49eba4252c2af4240442415b4abce3b87022a8f50"
            ],
            "markers": "python_version < '3.0'",
            "version": "==1.0.2"
        },
        "futures": {
            "hashes": [
                "sha256:9ec02aa7d674acb8618afb127e27fde7fc68994c0437ad759fa094a574adb265",
                "sha256:ec0a6cb848cc212002b9828c3e34c675e0c9ff6741dc445cab6fdd4e1085d1f1"
            ],
            "version": "==3.2.0"
        },
        "isort": {
            "hashes": [
                "sha256:1153601da39a25b14ddc54955dbbacbb6b2d19135386699e2ad58517953b34af",
                "sha256:b9c40e9750f3d77e6e4d441d8b0266cf555e7cdabdcff33c4fd06366ca761ef8",
                "sha256:ec9ef8f4a9bc6f71eec99e1806bfa2de401650d996c59330782b89a5555c1497"
            ],
            "version": "==4.3.4"
        },
        "lazy-object-proxy": {
            "hashes": [
                "sha256:0ce34342b419bd8f018e6666bfef729aec3edf62345a53b537a4dcc115746a33",
                "sha256:1b668120716eb7ee21d8a38815e5eb3bb8211117d9a90b0f8e21722c0758cc39",
                "sha256:209615b0fe4624d79e50220ce3310ca1a9445fd8e6d3572a896e7f9146bbf019",
                "sha256:27bf62cb2b1a2068d443ff7097ee33393f8483b570b475db8ebf7e1cba64f088",
                "sha256:27ea6fd1c02dcc78172a82fc37fcc0992a94e4cecf53cb6d73f11749825bd98b",
                "sha256:2c1b21b44ac9beb0fc848d3993924147ba45c4ebc24be19825e57aabbe74a99e",
                "sha256:2df72ab12046a3496a92476020a1a0abf78b2a7db9ff4dc2036b8dd980203ae6",
                "sha256:320ffd3de9699d3892048baee45ebfbbf9388a7d65d832d7e580243ade426d2b",
                "sha256:50e3b9a464d5d08cc5227413db0d1c4707b6172e4d4d915c1c70e4de0bbff1f5",
                "sha256:5276db7ff62bb7b52f77f1f51ed58850e315154249aceb42e7f4c611f0f847ff",
                "sha256:61a6cf00dcb1a7f0c773ed4acc509cb636af2d6337a08f362413c76b2b47a8dd",
                "sha256:6ae6c4cb59f199d8827c5a07546b2ab7e85d262acaccaacd49b62f53f7c456f7",
                "sha256:7661d401d60d8bf15bb5da39e4dd72f5d764c5aff5a86ef52a042506e3e970ff",
                "sha256:7bd527f36a605c914efca5d3d014170b2cb184723e423d26b1fb2fd9108e264d",
                "sha256:7cb54db3535c8686ea12e9535eb087d32421184eacc6939ef15ef50f83a5e7e2",
                "sha256:7f3a2d740291f7f2c111d86a1c4851b70fb000a6c8883a59660d95ad57b9df35",
                "sha256:81304b7d8e9c824d058087dcb89144842c8e0dea6d281c031f59f0acf66963d4",
                "sha256:933947e8b4fbe617a51528b09851685138b49d511af0b6c0da2539115d6d4514",
                "sha256:94223d7f060301b3a8c09c9b3bc3294b56b2188e7d8179c762a1cda72c979252",
                "sha256:ab3ca49afcb47058393b0122428358d2fbe0408cf99f1b58b295cfeb4ed39109",
                "sha256:bd6292f565ca46dee4e737ebcc20742e3b5be2b01556dafe169f6c65d088875f",
                "sha256:cb924aa3e4a3fb644d0c463cad5bc2572649a6a3f68a7f8e4fbe44aaa6d77e4c",
                "sha256:d0fc7a286feac9077ec52a927fc9fe8fe2fabab95426722be4c953c9a8bede92",
                "sha256:ddc34786490a6e4ec0a855d401034cbd1242ef186c20d79d2166d6a4bd449577",
                "sha256:e34b155e36fa9da7e1b7c738ed7767fc9491a62ec6af70fe9da4a057759edc2d",
                "sha256:e5b9e8f6bda48460b7b143c3821b21b452cb3a835e6bbd5dd33aa0c8d3f5137d",
                "sha256:e81ebf6c5ee9684be8f2c87563880f93eedd56dd2b6146d8a725b50b7e5adb0f",
                "sha256:eb91be369f945f10d3a49f5f9be8b3d0b93a4c2be8f8a5b83b0571b8123e0a7a",
                "sha256:f460d1ceb0e4a5dcb2a652db0904224f367c9b3c1470d5a7683c0480e582468b"
            ],
            "version": "==1.3.1"
        },
        "mccabe": {
            "hashes": [
                "sha256:ab8a6258860da4b6677da4bd2fe5dc2c659cff31b3ee4f7f5d64e79735b80d42",
                "sha256:dd8d182285a0fe56bace7f45b5e7d1a6ebcbf524e8f3bd87eb0f125271b8831f"
            ],
            "version": "==0.6.1"
        },
        "more-itertools": {
            "hashes": [
                "sha256:2b6b9893337bfd9166bee6a62c2b0c9fe7735dcf85948b387ec8cba30e85d8e8",
                "sha256:6703844a52d3588f951883005efcf555e49566a48afd4db4e965d69b883980d3",
                "sha256:a18d870ef2ffca2b8463c0070ad17b5978056f403fb64e3f15fe62a52db21cc0"
            ],
            "version": "==4.2.0"
        },
        "pluggy": {
            "hashes": [
                "sha256:7f8ae7f5bdf75671a718d2daf0a64b7885f74510bcd98b1a0bb420eb9a9d0cff",
                "sha256:d345c8fe681115900d6da8d048ba67c25df42973bda370783cd58826442dcd7c",
                "sha256:e160a7fcf25762bb60efc7e171d4497ff1d8d2d75a3d0df7a21b76821ecbf5c5"
            ],
            "version": "==0.6.0"
        },
        "py": {
            "hashes": [
                "sha256:29c9fab495d7528e80ba1e343b958684f4ace687327e6f789a94bf3d1915f881",
                "sha256:983f77f3331356039fdd792e9220b7b8ee1aa6bd2b25f567a963ff1de5a64f6a"
            ],
            "version": "==1.5.3"
        },
        "pylint": {
            "hashes": [
                "sha256:a48070545c12430cfc4e865bf62f5ad367784765681b3db442d8230f0960aa3c",
                "sha256:fff220bcb996b4f7e2b0f6812fd81507b72ca4d8c4d05daf2655c333800cb9b3"
            ],
            "index": "pypi",
            "version": "==1.9.2"
        },
        "pytest": {
            "hashes": [
                "sha256:26838b2bc58620e01675485491504c3aa7ee0faf335c37fcd5f8731ca4319591",
                "sha256:32c49a69566aa7c333188149ad48b58ac11a426d5352ea3d8f6ce843f88199cb"
            ],
            "index": "pypi",
            "version": "==3.6.1"
        },
        "singledispatch": {
            "hashes": [
                "sha256:5b06af87df13818d14f08a028e42f566640aef80805c3b50c5056b086e3c2b9c",
                "sha256:833b46966687b3de7f438c761ac475213e53b306740f1abfaa86e1d1aae56aa8"
            ],
            "markers": "python_version < '3.4'",
            "version": "==3.4.0.3"
        },
        "six": {
            "hashes": [
                "sha256:70e8a77beed4562e7f14fe23a786b54f6296e34344c23bc42f07b15018ff98e9",
                "sha256:832dc0e10feb1aa2c68dcc57dbb658f1c7e65b9b61af69048abc87a2db00a0eb"
            ],
            "version": "==1.11.0"
        },
        "wrapt": {
            "hashes": [
                "sha256:d4d560d479f2c21e1b5443bbd15fe7ec4b37fe7e53d335d3b9b0a7b1226fe3c6"
            ],
            "version": "==1.10.11"
        }
    }
}

```
</details>

------------

##### Expected result

Package installed correctly.

##### Actual result

<details><summary>AssertionError: This should match everything</summary>

```
Creating a virtualenv for this project…
Using /Users/yujunz/.pyenv/versions/2.7.13/bin/python2.7 (2.7.13) to create virtualenv…
⠋Already using interpreter /Users/yujunz/.pyenv/versions/2.7.13/bin/python2.7
New python executable in /Users/yujunz/.virtualenvs/pdm-cli-Acn4KsWS/bin/python2.7
Also creating executable in /Users/yujunz/.virtualenvs/pdm-cli-Acn4KsWS/bin/python
Installing setuptools, pip, wheel...done.

Virtualenv location: /Users/yujunz/.virtualenvs/pdm-cli-Acn4KsWS
Installing ../python-anyconfig…
⠙Installing u'../python-anyconfig'
$ "/Users/yujunz/.virtualenvs/pdm-cli-Acn4KsWS/bin/pip" install   --verbose    "../python-anyconfig" -i https://pypi.python.org/simple --exists-action w
Created temporary directory: /private/var/folders/tf/g7vrf5ns5ql_bv82wz9t_0rr0000gn/T/pip-ephem-wheel-cache-NgO1uM
Created temporary directory: /private/var/folders/tf/g7vrf5ns5ql_bv82wz9t_0rr0000gn/T/pip-install-usZraj
Looking in indexes: https://pypi.python.org/simple
Processing /Users/yujunz/Workspace/python-anyconfig
  Created temporary directory: /private/var/folders/tf/g7vrf5ns5ql_bv82wz9t_0rr0000gn/T/pip-req-build-5Aa6M4
  Running setup.py (path:/private/var/folders/tf/g7vrf5ns5ql_bv82wz9t_0rr0000gn/T/pip-req-build-5Aa6M4/setup.py) egg_info for package from file:///Users/yujunz/Workspace/python-anyconfig
    Running command python setup.py egg_info
    running egg_info
    creating pip-egg-info/anyconfig.egg-info
    writing pip-egg-info/anyconfig.egg-info/PKG-INFO
    writing top-level names to pip-egg-info/anyconfig.egg-info/top_level.txt
    writing dependency_links to pip-egg-info/anyconfig.egg-info/dependency_links.txt
    writing entry points to pip-egg-info/anyconfig.egg-info/entry_points.txt
    writing manifest file 'pip-egg-info/anyconfig.egg-info/SOURCES.txt'
    reading manifest file 'pip-egg-info/anyconfig.egg-info/SOURCES.txt'
    reading manifest template 'MANIFEST.in'
    warning: no files found matching '*.txt' under directory 'tests'
    writing manifest file 'pip-egg-info/anyconfig.egg-info/SOURCES.txt'
  Source in /private/var/folders/tf/g7vrf5ns5ql_bv82wz9t_0rr0000gn/T/pip-req-build-5Aa6M4 has version 0.9.4, which satisfies requirement anyconfig==0.9.4 from file:///Users/yujunz/Workspace/python-anyconfig
Building wheels for collected packages: anyconfig
  Created temporary directory: /private/var/folders/tf/g7vrf5ns5ql_bv82wz9t_0rr0000gn/T/pip-wheel-8MeBIU
  Running setup.py bdist_wheel for anyconfig: started
  Destination directory: /private/var/folders/tf/g7vrf5ns5ql_bv82wz9t_0rr0000gn/T/pip-wheel-8MeBIU
  Running command /Users/yujunz/.virtualenvs/pdm-cli-Acn4KsWS/bin/python2.7 -u -c "import setuptools, tokenize;__file__='/private/var/folders/tf/g7vrf5ns5ql_bv82wz9t_0rr0000gn/T/pip-req-build-5Aa6M4/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /private/var/folders/tf/g7vrf5ns5ql_bv82wz9t_0rr0000gn/T/pip-wheel-8MeBIU --python-tag cp27
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib
  creating build/lib/anyconfig
  copying anyconfig/__init__.py -> build/lib/anyconfig
  copying anyconfig/api.py -> build/lib/anyconfig
  copying anyconfig/backends.py -> build/lib/anyconfig
  copying anyconfig/cli.py -> build/lib/anyconfig
  copying anyconfig/compat.py -> build/lib/anyconfig
  copying anyconfig/dicts.py -> build/lib/anyconfig
  copying anyconfig/globals.py -> build/lib/anyconfig
  copying anyconfig/init.py -> build/lib/anyconfig
  copying anyconfig/parser.py -> build/lib/anyconfig
  copying anyconfig/query.py -> build/lib/anyconfig
  copying anyconfig/schema.py -> build/lib/anyconfig
  copying anyconfig/template.py -> build/lib/anyconfig
  copying anyconfig/utils.py -> build/lib/anyconfig
  running egg_info
  writing anyconfig.egg-info/PKG-INFO
  writing top-level names to anyconfig.egg-info/top_level.txt
  writing dependency_links to anyconfig.egg-info/dependency_links.txt
  writing entry points to anyconfig.egg-info/entry_points.txt
  reading manifest file 'anyconfig.egg-info/SOURCES.txt'
  reading manifest template 'MANIFEST.in'
  warning: no files found matching '*.txt' under directory 'tests'
  writing manifest file 'anyconfig.egg-info/SOURCES.txt'
  creating build/lib/anyconfig/backend
  copying anyconfig/backend/__init__.py -> build/lib/anyconfig/backend
  copying anyconfig/backend/base.py -> build/lib/anyconfig/backend
  copying anyconfig/backend/configobj.py -> build/lib/anyconfig/backend
  copying anyconfig/backend/ini.py -> build/lib/anyconfig/backend
  copying anyconfig/backend/json.py -> build/lib/anyconfig/backend
  copying anyconfig/backend/pickle.py -> build/lib/anyconfig/backend
  copying anyconfig/backend/properties.py -> build/lib/anyconfig/backend
  copying anyconfig/backend/shellvars.py -> build/lib/anyconfig/backend
  copying anyconfig/backend/toml.py -> build/lib/anyconfig/backend
  copying anyconfig/backend/xml.py -> build/lib/anyconfig/backend
  copying anyconfig/backend/yaml.py -> build/lib/anyconfig/backend
  warning: build_py: byte-compiling is disabled, skipping.

  installing to build/bdist.macosx-10.12-x86_64/wheel
  running install
  running install_lib
  creating build/bdist.macosx-10.12-x86_64
  creating build/bdist.macosx-10.12-x86_64/wheel
  creating build/bdist.macosx-10.12-x86_64/wheel/anyconfig
  copying build/lib/anyconfig/__init__.py -> build/bdist.macosx-10.12-x86_64/wheel/anyconfig
  copying build/lib/anyconfig/api.py -> build/bdist.macosx-10.12-x86_64/wheel/anyconfig
  creating build/bdist.macosx-10.12-x86_64/wheel/anyconfig/backend
  copying build/lib/anyconfig/backend/__init__.py -> build/bdist.macosx-10.12-x86_64/wheel/anyconfig/backend
  copying build/lib/anyconfig/backend/base.py -> build/bdist.macosx-10.12-x86_64/wheel/anyconfig/backend
  copying build/lib/anyconfig/backend/configobj.py -> build/bdist.macosx-10.12-x86_64/wheel/anyconfig/backend
  copying build/lib/anyconfig/backend/ini.py -> build/bdist.macosx-10.12-x86_64/wheel/anyconfig/backend
  copying build/lib/anyconfig/backend/json.py -> build/bdist.macosx-10.12-x86_64/wheel/anyconfig/backend
  copying build/lib/anyconfig/backend/pickle.py -> build/bdist.macosx-10.12-x86_64/wheel/anyconfig/backend
  copying build/lib/anyconfig/backend/properties.py -> build/bdist.macosx-10.12-x86_64/wheel/anyconfig/backend
  copying build/lib/anyconfig/backend/shellvars.py -> build/bdist.macosx-10.12-x86_64/wheel/anyconfig/backend
  copying build/lib/anyconfig/backend/toml.py -> build/bdist.macosx-10.12-x86_64/wheel/anyconfig/backend
  copying build/lib/anyconfig/backend/xml.py -> build/bdist.macosx-10.12-x86_64/wheel/anyconfig/backend
  copying build/lib/anyconfig/backend/yaml.py -> build/bdist.macosx-10.12-x86_64/wheel/anyconfig/backend
  copying build/lib/anyconfig/backends.py -> build/bdist.macosx-10.12-x86_64/wheel/anyconfig
  copying build/lib/anyconfig/cli.py -> build/bdist.macosx-10.12-x86_64/wheel/anyconfig
  copying build/lib/anyconfig/compat.py -> build/bdist.macosx-10.12-x86_64/wheel/anyconfig
  copying build/lib/anyconfig/dicts.py -> build/bdist.macosx-10.12-x86_64/wheel/anyconfig
  copying build/lib/anyconfig/globals.py -> build/bdist.macosx-10.12-x86_64/wheel/anyconfig
  copying build/lib/anyconfig/init.py -> build/bdist.macosx-10.12-x86_64/wheel/anyconfig
  copying build/lib/anyconfig/parser.py -> build/bdist.macosx-10.12-x86_64/wheel/anyconfig
  copying build/lib/anyconfig/query.py -> build/bdist.macosx-10.12-x86_64/wheel/anyconfig
  copying build/lib/anyconfig/schema.py -> build/bdist.macosx-10.12-x86_64/wheel/anyconfig
  copying build/lib/anyconfig/template.py -> build/bdist.macosx-10.12-x86_64/wheel/anyconfig
  copying build/lib/anyconfig/utils.py -> build/bdist.macosx-10.12-x86_64/wheel/anyconfig
  warning: install_lib: byte-compiling is disabled, skipping.

  running install_data
  creating build/bdist.macosx-10.12-x86_64/wheel/anyconfig-0.9.4.data
  creating build/bdist.macosx-10.12-x86_64/wheel/anyconfig-0.9.4.data/data
  creating build/bdist.macosx-10.12-x86_64/wheel/anyconfig-0.9.4.data/data/share
  creating build/bdist.macosx-10.12-x86_64/wheel/anyconfig-0.9.4.data/data/share/man
  creating build/bdist.macosx-10.12-x86_64/wheel/anyconfig-0.9.4.data/data/share/man/man1
  copying docs/anyconfig_cli.1 -> build/bdist.macosx-10.12-x86_64/wheel/anyconfig-0.9.4.data/data/share/man/man1
  running install_egg_info
  Copying anyconfig.egg-info to build/bdist.macosx-10.12-x86_64/wheel/anyconfig-0.9.4-py2.7.egg-info
  running install_scripts
  creating build/bdist.macosx-10.12-x86_64/wheel/anyconfig-0.9.4.dist-info/WHEEL
  creating '/private/var/folders/tf/g7vrf5ns5ql_bv82wz9t_0rr0000gn/T/pip-wheel-8MeBIU/anyconfig-0.9.4-cp27-none-any.whl' and adding '.' to it
  adding 'anyconfig/__init__.py'
  adding 'anyconfig/api.py'
  adding 'anyconfig/backends.py'
  adding 'anyconfig/cli.py'
  adding 'anyconfig/compat.py'
  adding 'anyconfig/dicts.py'
  adding 'anyconfig/globals.py'
  adding 'anyconfig/init.py'
  adding 'anyconfig/parser.py'
  adding 'anyconfig/query.py'
  adding 'anyconfig/schema.py'
  adding 'anyconfig/template.py'
  adding 'anyconfig/utils.py'
  adding 'anyconfig/backend/__init__.py'
  adding 'anyconfig/backend/base.py'
  adding 'anyconfig/backend/configobj.py'
  adding 'anyconfig/backend/ini.py'
  adding 'anyconfig/backend/json.py'
  adding 'anyconfig/backend/pickle.py'
  adding 'anyconfig/backend/properties.py'
  adding 'anyconfig/backend/shellvars.py'
  adding 'anyconfig/backend/toml.py'
  adding 'anyconfig/backend/xml.py'
  adding 'anyconfig/backend/yaml.py'
  adding 'anyconfig-0.9.4.data/data/share/man/man1/anyconfig_cli.1'
  adding 'anyconfig-0.9.4.dist-info/entry_points.txt'
  adding 'anyconfig-0.9.4.dist-info/top_level.txt'
  adding 'anyconfig-0.9.4.dist-info/WHEEL'
  adding 'anyconfig-0.9.4.dist-info/METADATA'
  adding 'anyconfig-0.9.4.dist-info/RECORD'
  removing build/bdist.macosx-10.12-x86_64/wheel
  Running setup.py bdist_wheel for anyconfig: finished with status 'done'
  Stored in directory: /Users/yujunz/Library/Caches/pip/wheels/e2/41/f3/1c3fa46139527175be64b58d8ec7ef69e853f427fd38953cc7
  Removing source in /private/var/folders/tf/g7vrf5ns5ql_bv82wz9t_0rr0000gn/T/pip-req-build-5Aa6M4
Successfully built anyconfig
Installing collected packages: anyconfig

  changing mode of /Users/yujunz/.virtualenvs/pdm-cli-Acn4KsWS/bin/anyconfig_cli to 755
Successfully installed anyconfig-0.9.4
Cleaning up...

Adding ../python-anyconfig to Pipfile's [packages]…
Installing -e…
⠹Installing u'-e'
Traceback (most recent call last):
  File "/Users/yujunz/.pyenv/versions/2.7.13/bin/pipenv", line 11, in <module>
    load_entry_point('pipenv==2018.5.18', 'console_scripts', 'pipenv')()
  File "/Users/yujunz/.pyenv/versions/2.7.13/lib/python2.7/site-packages/pipenv/vendor/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/Users/yujunz/.pyenv/versions/2.7.13/lib/python2.7/site-packages/pipenv/vendor/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/Users/yujunz/.pyenv/versions/2.7.13/lib/python2.7/site-packages/pipenv/vendor/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/yujunz/.pyenv/versions/2.7.13/lib/python2.7/site-packages/pipenv/vendor/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/yujunz/.pyenv/versions/2.7.13/lib/python2.7/site-packages/pipenv/vendor/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/Users/yujunz/.pyenv/versions/2.7.13/lib/python2.7/site-packages/pipenv/cli.py", line 402, in install
    selective_upgrade=selective_upgrade,
  File "/Users/yujunz/.pyenv/versions/2.7.13/lib/python2.7/site-packages/pipenv/core.py", line 1978, in do_install
    extra_indexes=extra_indexes,
  File "/Users/yujunz/.pyenv/versions/2.7.13/lib/python2.7/site-packages/pipenv/core.py", line 1420, in pip_install
    package_name.split('--hash')[0].split('--trusted-host')[0]
  File "/Users/yujunz/.pyenv/versions/2.7.13/lib/python2.7/site-packages/pipenv/utils.py", line 142, in get_requirement
    req = [r for r in requirements.parse(dep)][0]
  File "/Users/yujunz/.pyenv/versions/2.7.13/lib/python2.7/site-packages/pipenv/vendor/requirements/parser.py", line 50, in parse
    yield Requirement.parse(line)
  File "/Users/yujunz/.pyenv/versions/2.7.13/lib/python2.7/site-packages/pipenv/vendor/requirements/requirement.py", line 218, in parse
    re.sub(r'^(-e|--editable=?)\s*', '', line))
  File "/Users/yujunz/.pyenv/versions/2.7.13/lib/python2.7/site-packages/pipenv/vendor/requirements/requirement.py", line 123, in parse_editable
    assert local_match is not None, 'This should match everything'
AssertionError: This should match everything
/Users/yujunz/.pyenv/versions/2.7.13/lib/python2.7/site-packages/pipenv/_compat.py:86: ResourceWarning: Implicitly cleaning up <TemporaryDirectory '/var/folders/tf/g7vrf5ns5ql_bv82wz9t_0rr0000gn/T/pipenv-mvojaD-requirements'>
  warnings.warn(warn_message, ResourceWarning)
```
</details>

##### Steps to replicate

```
➜  pdm-cli git:(player) ✗ pipenv install ../python-anyconfig -e --verbose
```

**Comment by techalchemy** [comment]
The `-e` goes before the path, or it gets consumed as a part of whatever comes next on the command line.

`pipenv install -e ../python-anyconfig`

**Comment by uranusjr** [comment]
Likely a good idea to improve error messaging though.

**Comment by uranusjr** [comment]
This should be quite simple—replace the match anything part with an error message, and exit with return code 1.

**Comment by techalchemy** [comment]
Im pretty sure it fixes itself if we turn it into a click argument

**Comment by uranusjr** [comment]
@techalchemy Yeah but that’s been stalled for a while now too…

**Comment by techalchemy** [comment]
Closed via beb6aef7dd121fc6d3f6475e2e41111a7a18941c
